backend/handlers/handlers.go

```diff
@@ -339,6 +343,25 @@ func (handlers *Handlers) postAddAccountHandler(r *http.Request) (interface{}, e
 		return nil, err
 	}
 	keypath := signing.NewEmptyAbsoluteKeypath()
+
+	if jsonAddress != "" {
+		if !common.IsHexAddress(jsonAddress) {
```

Comment:
eth address check should happen in the `if jsonCoinCode == "teth" || jsonCoinCode == "eth" { ` branch